esp/esdllib/esdl_transformer2.cpp

```diff
@@ -895,7 +922,53 @@ void Esdl2Struct::process(Esdl2TransformerContext &ctx, IPropertyTree *pt, const
             {
                 const char *tagname = child.queryInputName(ctx);
                 if (pt->hasProp(tagname)||child.hasDefaults())
-                    child.process(ctx, pt->queryPropTree(tagname), child.queryOutputName(ctx), &local, count);
+                {
+                    unsigned numChildren = pt->getCount(tagname);
+                    bool isEsdlList = false;
+                    IEsdlDefObject*  def = child.queryEsdlDefObject();
+                    if (def->getEsdlType() == EsdlTypeArray)
+                    {
+                        IEsdlDefArray* defArray = dynamic_cast<IEsdlDefArray*>(def);
+                        isEsdlList = defArray->checkIsEsdlList();
+                    }
+                    if (!isEsdlList)
+                    {
+                        if (numChildren < 2)
+                            child.process(ctx, pt->queryPropTree(tagname), child.queryOutputName(ctx), &local, count);
+                        else
+                        {
```

Comment:
Why did this EsdlArray (non-list) behavior change? It looks like you are supporting multiple sections even for EsdlArrays now?

 ```
<root>
   <myarray>
     <item>a</item>
     <item>b</item>
     <item>c</item>
  </myarray>
  <tag1>www</tag1>
  <myarray>
     <item>x</item>
     <item>y</item>
     <item>z</item>
  </myarray>
</root>
```
?